Sources/SmileID/Classes/FaceDetector/FaceDetectorV2.swift

```diff
@@ -131,40 +162,39 @@ class FaceDetectorV2: NSObject {
 
         let faceRect = CGRect(origin: origin, size: size)
 
-        guard let croppedImage = cgImage.cropping(to: faceRect) else {
-            return nil
+        guard let croppedCGImage = cgImage.cropping(to: faceRect) else {
+            throw FaceDetectorError.unableToCropImage
         }
 
-        return UIImage(cgImage: croppedImage)
-    }
-}
+        let croppedImage = UIImage(cgImage: croppedCGImage)
+        guard let resizedImage = croppedImage.pixelBuffer(width: cropSize.width, height: cropSize.height) else {
+            throw FaceDetectorError.unableToCropImage
+        }
 
-// MARK: - Private methods
-extension FaceDetectorV2 {
-    func detectedFaceRectangles(request: VNRequest, error: Error?) {
-        guard let selfieViewModel = selfieViewModel,
-              let viewDelegate = viewDelegate else { return }
+        return resizedImage
+    }
 
-        guard let results = request.results as? [VNFaceObservation],
-                let result = results.first else {
-            selfieViewModel.perform(action: .noFaceDetected)
-            return
+    private func calculateBrightness(_ imageBuffer: CVPixelBuffer) -> Int {
+        guard let image = UIImage(pixelBuffer: imageBuffer),
+            let cgImage = image.cgImage,
```

Comment:
The functionalities are available in `CoreGraphics` that's why I needed to convert it to `CGImage` but I agree that it's possible to do the conversion in a single place and pass in the `cgImage` image instead of doing the conversion twice. Thanks for spotting that.